### PR TITLE
[RW-7544][risk: low] Add adminLocked state to database

### PR DESCRIPTION
**Pull request by dmohs**
Adds a boolean `adminLocked` to the workspace state and allows changing the value through an admin-accessible API endpoint.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/api/WorkspaceAdminController.java [hunk 1]
```diff
@@ -82,4 +86,13 @@ public WorkspaceAdminController(WorkspaceAdminService workspaceAdminService) {
...
   }
+
+  @Override
+  @AuthorityRequired({Authority.ACCESS_CONTROL_ADMIN})
+  public ResponseEntity<EmptyResponse> setAdminLockedState(
+      String workspaceNamespace, AdminLockedState lockedState) {
+    final boolean desiredLockState = Optional.ofNullable(lockedState.getValue()).orElse(false);
+    workspaceAdminService.setAdminLockedState(workspaceNamespace, desiredLockState);
+    return ResponseEntity.ok().build();
+  }
```
Comment (dmohs) [inline api/src/main/java/org/pmiops/workbench/api/WorkspaceAdminController.java:97]:
```suggestion

private boolean toPrimitive(Boolean value) {
  return Optional.ofNullable(value).orElse(false);
}

  @Override
  @AuthorityRequired({Authority.ACCESS_CONTROL_ADMIN})
  public ResponseEntity<EmptyResponse> setAdminLockedState(
      String workspaceNamespace, AdminLockedState lockedState) {
    workspaceAdminService.setAdminLockedState(workspaceNamespace, toPrimitive(lockedState.getValue()));
    return ResponseEntity.ok().build();
  }
```

Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/api/WorkspaceAdminController.java:97]: I like it

File: api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java [hunk 2]
```diff
@@ -314,6 +314,35 @@ public String getReadOnlyNotebook(
...
+  public void setAdminLockedState(String workspaceNamespace, boolean desiredLockState) {
+    // OK I got your request
+    log.info(
+        String.format(
+            "called setLockedState on wsns %s with desiredLockState %b",
+            workspaceNamespace, desiredLockState));
+
+    // TODO but what should I do with it?
+
+    Optional<DbWorkspace> dbWorkspaceOptional = getFirstWorkspaceByNamespace(workspaceNamespace);
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java:327]: this version was intended to be lenient for prototyping.  For the actual implementation I prefer the `getFirstWorkspaceByNamespaceOrThrow()` call.

File: api/src/test/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceTest.java [hunk 3]
```diff
@@ -45,6 +45,7 @@
 import org.pmiops.workbench.db.dao.UserService;
 import org.pmiops.workbench.db.dao.WorkspaceDao;
 import org.pmiops.workbench.db.model.DbWorkspace;
+import org.pmiops.workbench.exceptions.NotFoundException;
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceTest.java:48]: please revert

**Review by jmthibault79 — COMMENTED** [review]
Two main concerns:

We also need to implement the new field in Liquibase.

This does not cover the audit logging and additional parameter requirements (arguably these can wait for the next PR)

**Comment by jmthibault79** [comment]
👍 OK to wait for another PR for extra params / audit logging.  But I'm considering that a requirement for the **story**

**Comment by jmthibault79** [comment]
For fun, I thought I'd see what happens when trying to run this without a Liquibase change.

The API server starts just fine.  It fails when something touches DbWorkspace:
```
Nov 05, 2021 5:10:31 PM org.pmiops.workbench.exceptions.ExceptionAdvice serverError
SEVERE: ErrorId a75ab322-9f7c-4208-a280-3a9f94cce448: org.springframework.dao.InvalidDataAccessResourceUsageException
org.springframework.dao.InvalidDataAccessResourceUsageException: could not extract ResultSet; SQL [n/a]; nested exception is org.hibernate.exception.SQLGrammarException: could not extract ResultSet
...
Caused by: java.sql.SQLSyntaxErrorException: Unknown column 'dbworkspac0_.admin_locked' in 'field list'
```

**Review by jmthibault79 — APPROVED** [review]

**Pull request merged**